sklearn/pipeline.py

```diff
@@ -220,20 +236,44 @@ def _final_estimator(self):
 
     def _fit(self, X, y=None, **fit_params):
         self._validate_steps()
+        # Setup the memory
+        memory = self.memory
+        if memory is None:
+            memory = Memory(cachedir=None, verbose=0)
+        elif isinstance(memory, six.string_types):
+            memory = Memory(cachedir=memory, verbose=0)
+        elif not isinstance(memory, Memory):
+            raise ValueError('memory is either `str` or a `joblib.Memory`'
```

Comment:
Not sure that the markdown is really necessary ... I would do:

`"'memory' should either be a string or a joblib.Memory instance, got 'memory={!r}' instead.".format(memory)`